src/portable/renesas/link/link_type.h

```diff
@@ -0,0 +1,1624 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * This file is part of the TinyUSB stack.
+ */
+
+#ifndef _TUSB_LINK_TYPE_H_
+#define _TUSB_LINK_TYPE_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*--------------------------------------------------------------------*/
+/* Register Definitions                                           */
+/*--------------------------------------------------------------------*/
+
+/* Start of definition of packed structs (used by the CCRX toolchain) */
+TU_ATTR_PACKED_BEGIN
+TU_ATTR_BIT_FIELD_ORDER_BEGIN
+
+typedef struct TU_ATTR_PACKED {
+	union {
+		volatile uint16_t E; /* (@ 0x00000000) Pipe Transaction Counter Enable Register */
+
+		struct {
+			uint16_t : 8;
+			volatile uint16_t TRCLR : 1; /* [8..8] Transaction Counter Clear */
+			volatile uint16_t TRENB : 1; /* [9..9] Transaction Counter Enable */
+			uint16_t : 6;
+		} E_bits;
+	};
+
+	union {
+		volatile uint16_t N; /* (@ 0x00000002) Pipe Transaction Counter Register */
+
+		struct {
+			volatile uint16_t TRNCNT : 16; /* [15..0] Transaction Counter */
+		} N_bits;
+	};
+} LINK_REG_PIPE_TR_t; /* Size = 4 (0x4) */
+
+TU_VERIFY_STATIC(sizeof(LINK_REG_PIPE_TR_t) == 4, "incorrect size");
+
+/* LINK_REG Structure */
+typedef struct TU_ATTR_PACKED {
+	union {
+		volatile uint16_t SYSCFG; /* (@ 0x00000000) System Configuration Control Register */
+
+		struct {
+			volatile uint16_t USBE : 1; /* [0..0] USB Operation Enable */
+			uint16_t : 2;
+			volatile uint16_t DMRPU : 1; /* [3..3] D- Line Resistor Control */
+			volatile uint16_t DPRPU : 1; /* [4..4] D+ Line Resistor Control */
+			volatile uint16_t DRPD : 1;  /* [5..5] D+/D- Line Resistor Control */
+			volatile uint16_t DCFM : 1;  /* [6..6] Controller Function Select */
+			uint16_t : 1;
+			volatile uint16_t CNEN : 1; /* [8..8] CNEN Single End Receiver Enable */
+			uint16_t : 1;
+			volatile uint16_t SCKE : 1; /* [10..10] USB Clock Enable */
+			uint16_t : 5;
+		} SYSCFG_bits;
```

Comment:
Why not use the symbol name `SYSCFG_b` as defined in `renesas.h` .

The register definitions are very similar to those generated by [SVDConv](https://arm-software.github.io/CMSIS_5/SVD/html/svd_SVDConv_pg.html) with `--fields=struct`.

Reply:
was an effort to keep it vaguely similar to what was present before and readable, but in hindsight, the standard is better, will fix